### PR TITLE
 Consolidate duplicated methods from project and package controller 

### PR DESCRIPTION
**Pull request by krauselukas**
Due to the migration from bento to bootstrap, certain
methods turned unnecessary and can be removed.

### CODE REVIEW

**Comment by coolo** [comment]
Better not:

https://github.com/openSUSE/open-build-service/search?q=add_path&unscoped_q=add_path

**Comment by coolo** [comment]
at least your tests caught you

**Comment by dmarcoux** [comment]
The mixin depends on these methods, but doesn't define them, even though they are only used in the context of this mixin. The only difference between these methods is that the ones in `package_controller.rb` have `package: @package`. Checking if `@package` is not `nil` and then adding it can be done in the mixin. It would remove the duplication.

**Comment by dmarcoux** [comment]
One more thing, I just realized that `add_path` is only used in the Bento UI. So this can be removed, don't bother with refactoring this.

**Comment by coolo** [comment]
But the bootstrap tests fail after he removed it?

**Comment by dmarcoux** [comment]
I just updated my comment above with the cause of the failures. I'm still trying to see why it's redirecting.... it's weird.

**Comment by coolo** [comment]
High technology invented in 2012 :)

**Comment by krauselukas** [comment]
Seems like I opened the box of the pandora :D

**Comment by dmarcoux** [comment]
> High technology invented in 2012 :)

I'm trying to find when path/route helpers were introduced in Rails... Probably after 2012.. lol

**Comment by coolo** [comment]
Well, @dmarcoux is right about add_path though :)

And you can still merge users_path into the mixin

**Comment by coolo** [comment]
Important read here: https://en.wikipedia.org/wiki/Duck_typing

**Comment by dmarcoux** [comment]
> Important read here: https://en.wikipedia.org/wiki/Duck_typing

Quack :duck:

**Comment by dmarcoux** [comment]
Oh I see a fervent adapt of [rubber duck debugging](https://en.wikipedia.org/wiki/Rubber_duck_debugging)!

**Comment by krauselukas** [comment]
@dmarcoux okay, I think its ready for another review ;)

File: src/api/app/mixins/webui/manage_relationships.rb [hunk 1]
```diff
@@ -29,6 +23,10 @@ def save_person_or_group(what)
     end
   end
 
+  def users_path
+    url_for(action: :users, project: @package ? @project : @project.name, package: @package || nil)
```
Comment (coolo) [inline src/api/app/mixins/webui/manage_relationships.rb]: what do you think ` || nil` does to the value? :)

Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: why easy if there is also a comlicated solution :D thanks, long day ;)

File: src/api/app/mixins/webui/manage_relationships.rb [hunk 2]
```diff
@@ -1,12 +1,6 @@
 module Webui::ManageRelationships
-  def redirect_after_save(what)
-    if switch_to_webui2
-      redirect_to(users_path)
-    elsif what == :person
-      redirect_to(add_path(:add_person))
-    else
-      redirect_to(add_path(:add_group))
-    end
+  def redirect_after_save
```
Comment (coolo) [inline src/api/app/mixins/webui/manage_relationships.rb]: nitpick: that function is only used once and a one liner

Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: done

File: src/api/app/mixins/webui/manage_relationships.rb [hunk 3]
```diff
@@ -29,6 +19,10 @@ def save_person_or_group(what)
     end
   end
 
+  def users_path
```
Comment (dmarcoux) [inline src/api/app/mixins/webui/manage_relationships.rb]: Please rename this to something else to not overshadow the path helper method with the same name.

File: src/api/app/mixins/webui/manage_relationships.rb [hunk 4]
```diff
@@ -29,6 +19,10 @@ def save_person_or_group(what)
...
+    url_for(action: :users, project: @package ? @project : @project.name, package: @package)
```
Comment (dmarcoux) [inline src/api/app/mixins/webui/manage_relationships.rb]: Passing `@project` or `@project.name` generates the same URL, so this ternary operator isn't needed.

File: src/api/app/mixins/webui/manage_relationships.rb (hunk 3 above)
Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: done

File: src/api/app/mixins/webui/manage_relationships.rb (hunk 4 above)
Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: done

Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: Yeah :)

Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: Hmm weired doesn't show here that it is outdated

Comment (krauselukas) [reply src/api/app/mixins/webui/manage_relationships.rb]: Upps something got lost sorry

**Comment by dmarcoux** [comment]
Good job @krauselukas!

**Comment by krauselukas** [comment]
@dmarcoux merci :)

**Pull request merged**